apps/website/src/content/docs/tree.mdx

```diff
@@ -1,27 +1,149 @@
 ---
 title: Tree
-description: A tree provides a list of data.
+description: A tree provides a hierarchial lists of data with nested expandable levels.
 thumbnail: Tree
 ---
 
 <p>{frontmatter.description}</p>
 
-<Placeholder componentPageName='tree--basic' />
+## Usage
+
+A tree can be used to organize data in an application specific way, or it can be used to sort, filter, group, or search data as the user deems appropriate. To initialize the tree component, the following props are required:
+
+- `nodeRenderer`: A render function that returns the node element, which must be memoized. We recommend this function to return the `TreeNode` component.
+- `data`: An array of custom data used for [TreeNode](#treenode) inside `Tree`.
+- `getNode`: A function that maps your `data` entry to `NodeData` that has all information about the node state. We recommend controlling state of expanded, selected and disabled nodes via this prop. The function must be memoized.
 
 <LiveExample src='Tree.main.jsx'>
   <AllExamples.TreeMainExample client:load />
 </LiveExample>
 
-A tree can be used to organize data in an application specific way, or it can be used to sort, filter, group, or search data as the user deems appropriate. Each line of data level may begin with an eye icon for toggling its visibility. In this context, this icon is called specialty checkbox.
+Each level of the tree is composed of multiple `TreeNode` components, which can be used to display the node content. `TreeNode` must be used inside `Tree` component to correctly set node `depth` and `subNodes`.
+
+### Subnode
+
+The tree supports hierarchial data structures where each node can have subnodes, which can be expanded or collapsed. Subnodes allow handling nested data up to any desired depth.
+
+Each object in the `data` array can include an array of sub-item objects (e.g., `subItems`) to represent its children, enabling nested structures.
```

Comment:
It's not clear if `subItems` is an example or an API requirement. This could be rephrased to make it more clear.

Reply:
I have added more explanation.